// recastai-lib-browser/src/index.js
import { startRec, stopRec } from './recorde.js'
import textRequest from './textRequest.js'

(function () {
  class Recast {

    constructor(token, option) {
      if (!token) {
        console.log('no TOKEN find')
        return
      }
      this.webSocket = {
        ws: null,
        audioContext: null,
        audioInput: null,
        recorder: null,
        recording: false,
        audioTracks: null,
      }
      this.userToken = token
      if (option !== undefined) {
        this.debug = option.debug
        this.language = option.language
      }
    }
/*
*  Perform a speech request to Recast.AI
*  open a Websoket
*/
    startRecording() {
      if (!this.token) {
        return startRec(this)
      }
      return 'no token'
    }

/*
* Close the speed reqest to Recast.AI
* Close the Websoket
*/
    stopRecording() { stopRec(this) }

/*
* Perform a text request to Recast.AI
* @param {String} text: the text to process
*/
    textRequest(text) {
      if (!this.token) {
        return textRequest(text, this)
      }
      return 'no token'
    }
  }

  window.Recast = Recast
  module.exports = Recast
})()
